src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2848,4 +2972,206 @@ public function is_development_environment() {
 		return in_array( wp_get_environment_type(), array( 'development', 'local' ), true );
 	}
 
+	/**
+	 * Returns a list of headers and its verification callback to verify if page cache is enabled or not.
+	 *
+	 * Note: key is header name and value could be callable function to verify header value.
+	 * Empty value mean existence of header detect page cache is enabled.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return array List of client caching headers and their (optional) verification callbacks.
+	 */
+	public function get_page_cache_headers() {
+
+		$cache_hit_callback = static function ( $header_value ) {
+			return false !== strpos( strtolower( $header_value ), 'hit' );
+		};
+
+		return array(
+			'cache-control'          => static function ( $header_value ) {
+				return (bool) preg_match( '/max-age=[1-9]/', $header_value );
+			},
+			'expires'                => static function ( $header_value ) {
+				return strtotime( $header_value ) > time();
+			},
+			'age'                    => static function ( $header_value ) {
+				return is_numeric( $header_value ) && $header_value > 0;
+			},
+			'last-modified'          => '',
+			'etag'                   => '',
+			'x-cache'                => $cache_hit_callback,
+			'x-proxy-cache'          => $cache_hit_callback,
+			'cf-cache-status'        => $cache_hit_callback,
+			'x-kinsta-cache'         => $cache_hit_callback,
+			'x-aruba-cache'          => $cache_hit_callback,
+			'x-cache-enabled'        => static function ( $header_value ) {
+				return 'true' === strtolower( $header_value );
+			},
+			'x-cache-disabled'       => static function ( $header_value ) {
+				return ( 'on' !== strtolower( $header_value ) );
+			},
+			'cf-apo-via'             => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'tcache' );
+			},
+			'x-srcache-store-status' => $cache_hit_callback,
+			'x-srcache-fetch-status' => $cache_hit_callback,
+			'cf-edge-cache'          => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'cache' );
+			},
+		);
```

Comment:
Should there be a filter here for other hosting providers to add their own headers in?

Reply:
That's a good idea, I've added it "_page_cache_supported_cache_headers_"